catroid/src/main/java/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java

```diff
@@ -630,6 +630,7 @@ public boolean saveFormulaIfPossible() {
 				if (Brick.BrickField.isExpectingStringValue(currentBrickField)) {
 					return saveValidFormula(new FormulaElement(FormulaElement.ElementType.STRING, "", null));
 				}
+				// fallthrough
```

Comment:
Yes, this fallthrough is on purpose. 
If the BrickField we are editin the formula for is not expecting a string-value we want to return without saving when we get passed an empty string.